### PR TITLE
🐛 Make sure the interim output gets written

### LINKED ISSUE #356: if transform fails to create unique keys, it doesn't write interim output
```
ERROR - No unique keys were created for table! There must be at least 1 unique key column in a table. Zero unique keys in a table means there is no way to identify any concept instances. Source of error is /study/output/GuidedTransformStage/transform_function_output/default.tsv
```

That file hasn't been written yet.

### PR DESCRIPTION
**Pull request by fiendish**
Closes #356

### CODE REVIEW

File: kf_lib_data_ingest/etl/transform/transform.py [hunk 1]
```diff
@@ -415,9 +415,9 @@ def _insert_unique_keys(
         columns added to each DataFrame
         """
         self.logger.info('Begin unique key creation for standard concepts ...')
-        for extract_config_url, (source_file_url, df) in df_dict.items():
+        for reference, (df_name, df) in df_dict.items():
```
Comment (znatty22) [inline kf_lib_data_ingest/etl/transform/transform.py]: I understand why you're renaming extract_config_url to reference (bc of #357) , but what is df_name? Isn't this still source_file_url?

Comment (fiendish) [reply kf_lib_data_ingest/etl/transform/transform.py]: > Isn't this still source_file_url?

Nope. "Transform Module Output: default df"

Comment (znatty22) [reply kf_lib_data_ingest/etl/transform/transform.py]: Oh I see, this is for the intermediate data so there is no source file url 👍

Comment (fiendish) [reply kf_lib_data_ingest/etl/transform/transform.py]: Screen Shot 2019-07-19 at 10 43 48 AM

**Pull request merged**